src/stirling/source_connectors/perf_profiler/perf_profile_connector.cc

```diff
@@ -354,6 +360,27 @@ void PerfProfileConnector::ProcessBPFStackTraces(ConnectorContext* ctx, DataTabl
   profiler_state_->update_value(sample_count_idx, 0);
 }
 
+void PerfProfileConnector::CheckProfilerState() {
+  uint64_t error_code;
+  profiler_state_->get_value(kErrorStatusIdx, error_code);
+
+  DCHECK_EQ(error_code, kPerfProfilerStatusOk);
+
+  switch (error_code) {
+      case kOverflowError:
+        profiler_state_overflow_counter_.Increment();
+        break;
+      case kMapReadFailureError:
+        profiler_state_map_read_error_counter_.Increment();
+        break;
+  }
```

Comment:
I tried to add these prometheus metrics to a `std::array` so I could reduce the increment logic to a single line, but I ran into issues initializing the array and the prometheus library's deleted copy constructor.